expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -82,16 +83,31 @@ private static JsonNode addFields(String json, Publication publication) {
     }
 
     private static void expandLicenses(JsonNode node) {
-        var associatedArtifacts = node.get(ASSOCIATED_ARTIFACTS_FIELD);
-        if (nonNull(associatedArtifacts) && associatedArtifacts.isArray()) {
-            for (JsonNode artifact : associatedArtifacts) {
-                var artifactNode = (ObjectNode) artifact;
-                var licenseUri = extractLicenseFromAssociatedArtifactNode(artifact);
-                artifactNode.set(LICENSE_FIELD, License.fromUri(licenseUri).toJsonNode());
-            }
+        var optionalAssociatedArtifacts = Optional.ofNullable(node.get(ASSOCIATED_ARTIFACTS_FIELD));
+        optionalAssociatedArtifacts.ifPresent(ExpandedResource::handleAssociatedArtifacts);
+    }
+
+    private static void handleAssociatedArtifacts(JsonNode associatedArtifacts) {
+        if (associatedArtifacts.isArray()) {
+            associatedArtifacts.forEach(ExpandedResource::processArtifact);
         }
     }
 
+    private static void processArtifact(JsonNode artifact) {
+        if (isNotALink(artifact)) {
+            var artifactNode = (ObjectNode) artifact;
+            var licenseUri = extractLicenseFromAssociatedArtifactNode(artifact);
+            Optional.ofNullable(licenseUri)
+                .map(License::fromUri)
+                .map(License::toJsonNode)
+                .ifPresent(jsonNode -> artifactNode.set(LICENSE_FIELD, jsonNode));
+        }
+    }
```

Comment:
AssociatedArtifacts contains both Files and AssociatedLinks, we expand licenses only for Files.